src/app/leverage/components/your-tokens.tsx

```diff
@@ -26,17 +29,17 @@ export function YourTokens() {
 
   useEffect(() => {
     if (balances.length === 0) return
-    fetchLeverageTokenPrices(balances, setTokens)
-  }, [balances])
+    fetchLeverageTokenPrices(balances, setTokens, chainId ?? 1)
+  }, [balances, chainId])
 
   const handleCloseClick = (token: EnrichedToken) => {
     if (isMinting) toggleIsMinting()
 
-    const leverageBaseToken = getLeverageBaseToken(token)
+    const leverageBaseToken = getLeverageBaseToken(token.symbol)
     if (leverageBaseToken === null) return
     onSelectBaseToken(leverageBaseToken.symbol)
 
-    const leverageType = getLeverageType(token)
+    const leverageType = getLeverageType(token.symbol)
```

Comment:
Made all these functions to use only the symbol as they didn't really usy anything else from `Token`